File: java/src/test/java/problems201_300/InvertBinaryTreeTest.java
package problems201_300;

import concepts.TreeNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class InvertBinaryTreeTest extends SampledTest {
    @Test
    public void test() {
        for (InvertBinaryTree solution : InvertBinaryTree.values()) {
            for (InvertBinaryTreeSample sample : getSamples(InvertBinaryTreeSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.invertTree(TreeNode.of(sample.input.root)))
                    .isEqualTo(TreeNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/InvertBinaryTreeTest.kt
package problems201_300

import com.google.common.truth.Truth.assertWithMessage
import concepts.TreeNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class InvertBinaryTreeTest : SampledTest() {
    @Test
    fun test() =
        InvertBinaryTree.entries.forEach { solution ->
            getSamples<Array<InvertBinaryTreeSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.invertTree(TreeNode.of(*input.root)))
                    .isEqualTo(TreeNode.of(*output))
            }
        }
}
